src/orange/OrangeTrackView.hh

```diff
@@ -203,13 +204,23 @@ OrangeTrackView::operator=(const Initializer_t& init)
     local.temp_sense = this->make_temp_sense();
 
     // Initialize logical state
-    auto tracker = this->make_tracker(UniverseId{0});
-    auto tinit   = tracker.initialize(local);
-    // TODO: error correction/graceful failure if initialiation failured
-    CELER_ASSERT(tinit.volume && !tinit.surface);
+    UniverseId          uid{0};
+    VolumeId            global_vol_id{};
+    detail::UnitIndexer unit_indexer(params_.unit_indexer_data);
+
+    do
```

Comment:
A code comment at the beginning of the loop would be useful:
```suggestion
    // Recurse into daughter universes starting with the outermost universe
    do
```

Also instead of the "magic number" `UniverseId uid{0};` (which looks like it could be just a default initialization) we should add to `OrangeTypes.hh` (probably underneath `no_intersection`:
```c++
CELER_CONSTEXPR_FUNCTION UniverseId global_universe_id()
{
    return UniverseId{0};
}
```
and use that.

Reply:
I think this is confusing. We have local and global volume ids and surface ids, so this function gives the impression of global (e.g. non-local) "universe id" (which doesn't make sense of course), rather than "universe id of the highest-level universe". I suggest we call this "base_universe_id".

Reply:
Oh I see what you mean... "global" meaning top-level (level zero universe, least deep universe, root of the universe hierarchy, ...) is confusing because of the "local"/"global" naming convention. That's fair. I'm not sure about `base` since that's generally reserved for C++ hierarchy.

The ORANGE input file (following GG and KENO conventions) define a `global` keyword to indicate the top level universe. Maybe `top_universe_id`? And it's probably OK if the user interface uses "global" since they don't really have to be aware of the multi-universe indexing.

Reply:
I will use `top`.

Reply:
I think the name clash with `base` is much less confusing than the name clash with `global` though FTR.